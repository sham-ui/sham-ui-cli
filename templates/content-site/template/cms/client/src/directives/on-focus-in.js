import { EventListener } from 'sham-ui-directives';

export class onFocusIn extends EventListener {
    constructor() {
        super( 'focusin' );
    }
}
