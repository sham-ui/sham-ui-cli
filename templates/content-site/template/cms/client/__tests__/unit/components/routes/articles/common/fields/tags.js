import * as directives from 'sham-ui-directives';
import { onFocusIn } from '../../../../../../../src/directives/on-focus-in';
// eslint-disable-next-line max-len
import RoutesArticlesCommonFieldsTags  from '../../../../../../../src/components/routes/articles/common/fields/tags.sfc';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( RoutesArticlesCommonFieldsTags, {
        all: [
            { name: 'first' },
            { name: 'second' }
        ],
        directives: {
            onFocusIn,
            ...directives
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'show all items on click', () => {
    const meta = renderer( RoutesArticlesCommonFieldsTags, {
        all: [
            { name: 'first' },
            { name: 'second' }
        ],
        directives: {
            onFocusIn,
            ...directives
        }
    } );
    document.querySelector( 'input' ).click();
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'add tag', () => {
    const onChange = jest.fn();
    const tags = [
        { name: 'first' },
        { name: 'second' }
    ];
    const meta = renderer( RoutesArticlesCommonFieldsTags, {
        all: tags,
        selected: [],
        onChange,
        directives: {
            onFocusIn,
            ...directives
        }
    } );
    document.querySelector( 'input' ).click();
    document.querySelector( '[data-test-item="second"]' ).click();
    expect( meta.toJSON() ).toMatchSnapshot();
    expect( onChange ).toHaveBeenCalledTimes( 1 );
    expect( onChange.mock.calls[ 0 ] ).toHaveLength( 1 );
    expect( onChange.mock.calls[ 0 ][ 0 ] ).toEqual( [ tags[ 1 ] ] );
} );


it( 'remove tag', () => {
    const onChange = jest.fn();
    const tags = [
        { name: 'first' },
        { name: 'second' }
    ];
    const meta = renderer( RoutesArticlesCommonFieldsTags, {
        all: tags,
        selected: [ ...tags ],
        onChange,
        directives: {
            onFocusIn,
            ...directives
        }
    } );
    document.querySelector( '[data-test-remove-tag="second"]' ).click();
    expect( meta.toJSON() ).toMatchSnapshot();
    expect( onChange ).toHaveBeenCalledTimes( 1 );
    expect( onChange.mock.calls[ 0 ] ).toHaveLength( 1 );
    expect( onChange.mock.calls[ 0 ][ 0 ] ).toEqual( [ tags[ 0 ] ] );
} );
