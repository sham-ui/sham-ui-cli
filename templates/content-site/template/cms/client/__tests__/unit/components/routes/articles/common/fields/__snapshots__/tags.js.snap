// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add tag 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "all": Array [
      Object {
        "name": "first",
      },
      Object {
        "name": "second",
      },
    ],
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "onChangeUpdate": [Function],
      "onFocusIn": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "onChange": [MockFunction] {
      "calls": Array [
        Array [
          Array [
            Object {
              "name": "second",
            },
          ],
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    },
    "selected": Array [
      Object {
        "name": "second",
      },
    ],
  },
  "html": "
<div class=\\"tags-input\\">
  <!--0-->
  <div class=\\"autocomplete-input\\">
    <input autocomplete=\\"off\\">
    <!--0-->
  </div>
  <!--1-->
</div>
",
}
`;

exports[`remove tag 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "all": Array [
      Object {
        "name": "first",
      },
      Object {
        "name": "second",
      },
    ],
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "onChangeUpdate": [Function],
      "onFocusIn": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "onChange": [MockFunction] {
      "calls": Array [
        Array [
          Array [
            Object {
              "name": "first",
            },
          ],
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    },
    "selected": Array [
      Object {
        "name": "first",
      },
    ],
  },
  "html": "
<div class=\\"tags-input\\">
  <div class=\\"tag\\">first<span class=\\"remove\\" data-test-close-button=\\"\\" data-test-remove-tag=\\"first\\">×</span></div>
  <div class=\\"tag\\">second<span class=\\"remove\\" data-test-close-button=\\"\\" data-test-remove-tag=\\"second\\">×</span></div>
  <!--0-->
  <div class=\\"autocomplete-input\\">
    <input autocomplete=\\"off\\">
    <!--0-->
  </div>
  <!--1-->
</div>
",
}
`;

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "all": Array [
      Object {
        "name": "first",
      },
      Object {
        "name": "second",
      },
    ],
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "onChangeUpdate": [Function],
      "onFocusIn": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "onChange": [Function],
    "selected": Array [],
  },
  "html": "
<div class=\\"tags-input\\">
  <!--0-->
  <div class=\\"autocomplete-input\\">
    <input autocomplete=\\"off\\">
    <!--0-->
  </div>
  <!--1-->
</div>
",
}
`;

exports[`show all items on click 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "all": Array [
      Object {
        "name": "first",
      },
      Object {
        "name": "second",
      },
    ],
    "directives": Object {
      "EventListener": [Function],
      "UpdateOnEvent": [Function],
      "disabled": [Function],
      "onChangeUpdate": [Function],
      "onFocusIn": [Function],
      "onInputUpdate": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "onChange": [Function],
    "selected": Array [],
  },
  "html": "
<div class=\\"tags-input\\">
  <!--0-->
  <div class=\\"autocomplete-input\\">
    <input autocomplete=\\"off\\">
    <div class=\\"autocomplete-results\\">
      <div class=\\"result-item\\" data-test-item=\\"first\\">first</div>
      <div class=\\"result-item\\" data-test-item=\\"second\\">second</div>
    </div>
    <!--0-->
  </div>
  <!--1-->
</div>
",
}
`;
